Source/Wpf/Prism.Wpf/Services/Dialogs/DialogService.cs

```diff
@@ -126,10 +127,14 @@ void ConfigureDialogWindowProperties(IDialogWindow window, FrameworkElement dial
             window.Content = dialogContent;
             window.DataContext = viewModel; //we want the host window and the dialog to share the same data contex
 
-            //TODO: is there a better way to set the owner
-            window.Owner = Application.Current.Windows.OfType<Window>().FirstOrDefault(x => x.IsActive);
+            window.Owner = Application.Current?.GetActiveWindow(); // Application may be null if IsInDesignMode.
```

Comment:
I'm not sure if this is better, and introduces the use of a DLLImport calling pInvoke. Can you explain why this would be the preferred approach? Have you tested to see if this is .NET Core 3 compliant.

Reply:
Also this code could never be invoked during design time since a dialog is only created in response to an instance of the service being created and a ShowDialog being invoked. It doesn't hurt to check for null, but that should never be the case.

Reply:
(Almost) the same approach microsoft uses in order to resolve the owner of the default MessageBox window. Generally speaking .NET Core is compatible with DLLImport attribute and even corefx use it  (https://github.com/dotnet/corefx/search?q=dllimport&unscoped_q=dllimport). But you are completely right that i have to test it and maybe change it a little bit in order to be more generic.

As far as your question if this would be the preferred approach, i cannot answer with certainty. The cost of calling a native function is for sure high, but i am not sure how Windows.IsActive works.